### DISCUSSION TITLE
UI improvements for bookmarks

### DISCUSSION THREAD

**Discussion by tksoh** [eteran/nedit-ng#216] [Ideas]
Right now, bookmarks are invisible and users have to keep track of them manually, which make bookmarks kind of handicap. Few thing we can do:

1. add marker on the line number areas to show the bookmarks
2. add a dialog that hold the list of bookmarks and let user navigate to selected bookmark
3. add keyboard shortcut to traverse through the bookmarks

TK

**Comment by eteran** [comment]
All 2 are cool ideas. 

Because of how I implemented the number side pane, it probably wouldn't be hard to add extra icons to that too... a perfect place for bookmark indicators ;-)

**Comment by tksoh** [comment]
More ideas:

1. add context menu on line number area to add bookmarks. Some editors do it through a simple click, but I feel context menu is more organized, and allow future extension for more functionality
2. auto assign letters to bookmarks added with the context menu. I believe most of the time users don't really care which letter. Although nedit's unique use of bookmark letters has a nice effect of letting us go back to our 'favorite' bookmarks when needed. 
3. display the bookmark letters on the indicators (so we can remember our 'favorites' ;-), though this probably will be limited to the 26 letters that nedit is currently supporting.
